src/reporting/Semgrep_error_code.ml

```diff
@@ -108,56 +154,28 @@ let known_exn_to_error ?(rule_id = None) file (e : Exception.t) : error option =
         | Tok.OriginTok { str; _ } -> spf "`%s` was unexpected" str
         | __else__ -> "unknown reason"
       in
-      Some (mk_error_tok tok msg Out.ParseError)
+      Some (mk_error_tok rule_id tok msg Out.ParseError)
   | Parsing_error.Other_error (s, tok) ->
-      Some (mk_error_tok ~rule_id tok s Out.SpecifiedParseError)
-  | R.Err err -> (
-      match err with
```

Comment:
It moved to the function `opt_error_of_rule_error` above.

Reply:
The exception was renamed `Rule.Err` -> `Rule.Error` because the other `Error` constructor occurring in another type is elegantly disambiguate with a type annotation.

In general, use this:
```
match (x : Foo.bar) with
| Error _ -> ...
```
rather than
```
match x with
| Foo.Error _ -> ...
```
because the latter can be ambiguous and misses an opportunity to tell the reader what type we're dealing with.